Source/Scene/ClippingPlanesCollection.js

```diff
@@ -0,0 +1,189 @@
+define([
+    '../Core/BoundingSphere',
+    '../Core/buildModuleUrl',
+    '../Core/Cartesian3',
+    '../Core/Cartographic',
+    '../Core/Color',
+    '../Core/defaultValue',
+    '../Core/defined',
+    '../Core/Intersect',
+    '../Core/Matrix4',
+    '../Core/Plane'
+], function(
+    BoundingSphere,
+    buildModuleUrl,
+    Cartesian3,
+    Cartographic,
+    Color,
+    defaultValue,
+    defined,
+    Intersect,
+    Matrix4,
+    Plane) {
+    'use strict';
+
+    /**
+     * Contains the options to specify a set of clipping planes. Clipping planes selectively disable rendering on an object on the outside of the specified list of {@link Plane}.
+     *
+     * @alias ClippingPlanesCollection
+     * @constructor
+     *
+     * @param {Object} [options] Object with the following properties:
+     * @param {Plane[]} [options.planes=[]] An array of up to 6 {@link Plane} used to selectively disable rendering on the outside of each plane.
+     * @param {Boolean} [options.enabled=true] Determines whether the clipping planes are active.
+     * @param {Matrix4} [options.transformationMatrix=Matrix4.IDENTITY] The 4x4 transformation matrix specifying an additional transform relative to the clipping planes original coordinate system.
```

Comment:
> transformationMatrix

Not sure about this name.  Basically there is `WORLD`, `MODEL`, and `PLANE` coordinates, and the transform is:
```
P-in-WORLD = [MODEL-to-WORLD] * [PLANE-to-MODEL] * P-in-PLANE
```
where `MODEL-to-WORLD` is the typical `modelMatrix` on the primitive.

I think this could also be called `modelMatrix` since that is how it behaves if the plane were a primitive.